extension/options/src/App.js

```diff
@@ -0,0 +1,140 @@
+import React, { useEffect, useState } from "react";
+
+const defaultOptions = {
+  arrowParens: "avoid",
+  bracketSpacing: true,
+  jsxBracketSameLine: false,
+  printWidth: 80,
+  semi: true,
+  singleQuote: false,
+  tabWidth: 2,
+  trailingComma: "none",
+  useTabs: false
+};
+
+function App() {
+  const [options, setOptions] = useState();
+
+  useEffect(() => {
+    window.chrome.storage.sync.get(defaultOptions, setOptions);
```

Comment:
Seems like we only want to `chrome.storage.sync.get` this data on initial mount, and then we want to `chrome.storage.sync.set` every time time `options` changes (we can rely on this component's internal state for the view after the initial load). What does watching for `setOptions` do here? I'm new to hooks, so might be misunderstanding something :)

Reply:
`eslint-config-react-app` (part of CRA) complains if values are used in `useEffect` without being added to the dependencies array. I'm not sure if it really matters in this case: because the `setOptions` reference is stable, it will only run on the initial mount anyway (as if the dependencies were `[]` or you were using `componentDidMount` in a class component).

Reply:
Got it, thanks! I think it’s clearer to have it be an empty array, but I’m also new to hooks and defer to your judgment here.

Reply:
Whoops I must have been thinking about some previous code, it does actually allow the dependencies here to be `[]` because it knows `setOptiosn` is stable, updated.